rootsanalyzer/rootset_id.go

```diff
@@ -0,0 +1,60 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package rootsanalyzer
+
+import (
+	"crypto/sha256"
+	"fmt"
+	"sort"
+	"strings"
+
+	"github.com/google/certificate-transparency-go/x509"
+	"github.com/google/monologue/storage"
+)
+
+// SHAIDMaker is SHA256-based implementation IDMaker for root certificates ID generation.
+type SHAIDMaker struct {
+}
+
+func (m *SHAIDMaker) GenerateCertID(root *x509.Certificate) ([32]byte, error) {
```

Comment:
Where do you intend on calling this from (i.e. which package)?

Reply:
From any package that operates on those IDs

Reply:
Any idea what packages that'll be? I'd have thought only storage implementations would have use for this, in which case it probably doesn't belong in rootsanalyzer.

Reply:
I'm also wondering whether anything will need to call this func directly - does it need to be exported?